src/ucp/api/ucp.h

```diff
@@ -1346,6 +1349,21 @@ ucs_status_t ucp_worker_arm(ucp_worker_h worker);
 
 /**
  * @ingroup UCP_WAKEUP
+ * @brief Clear events from the worker file descriptor.
+ *
+ * This routine removes events from the worker file descriptor, so it will not
+ * be signaled any more. If this function is not called, the file descriptor
+ * remains in signaled state.
+ *
+ * @param [in]  worker    Worker to clear events on.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_worker_clear_efd(ucp_worker_h worker);
```

Comment:
cleaning events is usually a system call (because file descriptor state has to be update). so didn't want to add such overhead to ucp_worker_progress().